### PR TITLE
Add OpenShift deployment settings section

### PR DESCRIPTION
**Pull request by asaunier**

### CODE REVIEW

File: doc/integrator/docker.rst [hunk 1]
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
...
+
+Configure project for OpenShift
+-------------------------------
+
+This section is specific to projects using OpenShift for deployment, relying
+on Travis to build Docker images that are then pushed to Docker Hub.
+
+Edit file `.travis.yml` and update the environment variables:
+
+.. prompt:: bash
```
Comment (asaunier) [inline doc/integrator/docker.rst]: Using another format would make sense but "yaml" or even "text" make sphinx crash on my laptop!?

File: doc/integrator/docker.rst [hunk 2]
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
...
+
+    env:
+      global:
+        - PROJECT=<package name>
+        - OPENSHIFT_PROJECT=gs-gmf-<package name>
+        - HELM_RELEASE_NAMES=int,prod
+        - OPENSHIFT_URL=https://api.openshift-ch-1.camptocamp.com/
+        - DOCKER_USERNAME=<FIXME>
```
Comment (asaunier) [inline doc/integrator/docker.rst]: OK to publish the actual username here?

File: doc/integrator/docker.rst [hunk 3]
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
...
+        - secure: <encrypted password>
+
+The encrypted password is obtained using the ``travis encrypt`` command from
+within the project directory:
+
+.. prompt:: bash
+
+    cd <project directory>
+    gopass gs/ci/dockerhub/password
```
Comment (asaunier) [inline doc/integrator/docker.rst]: Perhaps a doc about gopass would be useful somewhere?

File: doc/integrator/docker.rst [hunk 4]
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
...
+    travis encrypt DOCKER_PASSWORD=<paste password returned by the line above>
+
+Make sure to add line breaks (escaping with ``\``) when filling the ``secure``
+argument so that line length constraints are fullfilled.
```
Comment (asaunier) [inline doc/integrator/docker.rst]: Is there a way to do it automatically or should the lines be broken by hand?

**Comment by asaunier** [comment]
@jwkaltz said

> If we could make information like "Clone https://github.com/camptocamp/private-geo-charts/ and create file" a bit generic, I think it would make sense to add all this to the official doc.Something like a section "Best practices for OpenShift", we recommend to have a repository named for example "private-geo-charts" which contains blablabla
> (but we do not mention the camptocamp private repository explicitly)

File: doc/integrator/docker.rst [hunk 5]
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
 
 * List the images with ``docker images``.
 * Remove the unwanted images with ``docker rmi <image>:<tag>``.
+
+
+Configure project for OpenShift
+-------------------------------
+
+This section is specific to projects using OpenShift for deployment, relying
+on Travis to build Docker images that are then pushed to Docker Hub.
```
Comment (sbrunner) [inline doc/integrator/docker.rst]: Shouldn't we add a note to informs about that can be done only by Camptocamp people?

File: doc/integrator/docker.rst [hunk 6]
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
 
 * List the images with ``docker images``.
 * Remove the unwanted images with ``docker rmi <image>:<tag>``.
+
+
+Configure project for OpenShift
```
Comment (sbrunner) [inline doc/integrator/docker.rst]: Configure Docker Hub publishing and OpenShift notifications

File: doc/integrator/docker.rst (hunk 2 above)
Comment (asaunier) [reply doc/integrator/docker.rst]: It's already in the scaffolds: https://github.com/camptocamp/c2cgeoportal/blob/2.4/geoportal/c2cgeoportal_geoportal/scaffolds/create/%2Bdot%2Btravis.yml_tmpl even though only commented out. I guess we can leave it as it.

Do you think it is OK if I actually replace ``<FIXME>`` by the real username C2C uses? Or would it be a security issue?

File: doc/integrator/docker.rst (hunk 3 above)
Comment (asaunier) [reply doc/integrator/docker.rst]: OK, but I don't remember what you did and have no link :P

**Comment by asaunier** [comment]
@sbrunner PR updated according to your comments.

I am a bit confused by the error reported by Jenkins (really related to my changes?):
> groovy.lang.MissingMethodException: No signature of method: org.jenkinsci.plugins.workflow.support.steps.build.RunWrapper.getBuildCauses() is applicable for argument types: () values: []

File: doc/integrator/docker.rst (hunk 2 above)
Comment (sbrunner) [reply doc/integrator/docker.rst]: No, he is already visible in the demo project :-)

File: doc/integrator/docker.rst (hunk 3 above)
Comment (sbrunner) [reply doc/integrator/docker.rst]: https://github.com/camptocamp/geospatial-ci-pass/blob/master/README.md :-)

**Comment by sbrunner** [comment]
:-/ I just re-run the job :-)

**Comment by asaunier** [comment]
> I just re-run the job

Still failing...
But anyway I guess there will be other additions to this doc. I'll see at the end if we still have the problem with Jenkins.

By the way I have made changes according to your last comments: 3162677

**Comment by asaunier** [comment]
@sbrunner I have updated the branch and fixed a few spelling errors reported by the checker. Now Jenkins passes.
The docker/OS doc can probably be improved but is it OK for you to merge the PR as is?

**Comment by sbrunner** [comment]
+1 :-)

**Pull request merged**